### ISSUE TITLE
[no-unused-vars] (v4-alpha.9) Incorrect warning about unused type with abstract field

### ISSUE THREAD

**Issue by eamodio** [typescript-eslint/typescript-eslint#2417]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-unused-vars": [
		"warn",
		{
			"args": "after-used",
			"argsIgnorePattern": "^_",
			"ignoreRestSiblings": true,
			"varsIgnorePattern": "^_$"
		}
	]
  }
}
```

fileA
```TS
export type FooType = 'foo' | 'bar';
```

fileB
```TS
import { FooType } from './fileA';

abstract class Foo {
	protected abstract readonly type: FooType;
}
```

**Expected Result**

 No errors/warnings

**Actual Result**
```
WARNING in fileB.ts:1:8
    @typescript-eslint/no-unused-vars: 'FooType' is defined but never used. Allowed unused vars must match /^_$/u.
```

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.0.0-alpha.9` |
| `@typescript-eslint/parser`        | `4.0.0-alpha.9` |
| `TypeScript`                       | `4.0.2` |
| `ESLint`                           | `7.7.0` |
| `node`                             | `12.16.2` |
